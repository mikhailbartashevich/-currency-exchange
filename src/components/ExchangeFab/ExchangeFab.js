import React from 'react';
import Fab from '@material-ui/core/Fab';
import { withStyles } from '@material-ui/styles';
import { pure } from 'recompose';

const styles = {
  root: {
    background: '#d80091',
    width: '100%',
    height: '30px',
    color: 'white',
    textTransform: 'none',
  },
};

const ExchangeFabRaw = props => {
  const { classes, color, ...other } = props;
  return (
    <Fab
      color="secondary"
      className={classes.root}
      classes={{ disabled: classes.disabled }}
      {...other}
    />
  );
};

export const ExchangeFab = pure(withStyles(styles)(ExchangeFabRaw));
