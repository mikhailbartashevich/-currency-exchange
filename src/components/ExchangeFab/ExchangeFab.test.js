import React from 'react';
import { shallow } from 'enzyme';
import { ExchangeFab } from './ExchangeFab';

describe('ExchangeFab', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<ExchangeFab debug />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with no props', () => {
    const component = shallow(<ExchangeFab />);
    expect(component).toMatchSnapshot();
  });

});
