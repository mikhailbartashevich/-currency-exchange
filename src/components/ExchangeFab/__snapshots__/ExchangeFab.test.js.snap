// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeFab should render correctly in "debug" mode 1`] = `
<ExchangeFabRaw
  classes={
    Object {
      "root": "ExchangeFabRaw-root-1",
    }
  }
  debug={true}
/>
`;

exports[`ExchangeFab should render correctly with no props 1`] = `
<ExchangeFabRaw
  classes={
    Object {
      "root": "ExchangeFabRaw-root-1",
    }
  }
/>
`;
